### PR TITLE
gambit-scheme 4.7.8

### PR DESCRIPTION
**Pull request by nijikon**

### CODE REVIEW

**Comment by DomT4** [comment]
```
gambit-scheme:
 * Emacs Lisp files were linked directly to /usr/local/share/emacs/site-lisp

This may cause conflicts with other packages; install to a subdirectory instead, such as
/usr/local/Cellar/gambit-scheme/4.7.8/share/emacs/site-lisp/gambit-scheme

The offending files are:
  /usr/local/Cellar/gambit-scheme/4.7.8/share/emacs/site-lisp/gambit.el
```

There should be a configure flag for this. @dunn's the Emacs guru :smiley:

**Comment by dunn** [comment]
Looks like it's set in the makefile.  This should work:
```diff
diff --git a/Library/Formula/gambit-scheme.rb b/Library/Formula/gambit-scheme.rb
index df89310..8c29a88 100644
--- a/Library/Formula/gambit-scheme.rb
+++ b/Library/Formula/gambit-scheme.rb
@@ -44,11 +44,11 @@ class GambitScheme < Formula
     end
 
     system "./configure", *args
-    system "make check" if build.with? "check"
+    system "make", "check" if build.with? "check"
 
     ENV.j1
     system "make"
-    system "make", "install"
+    system "make", "install", "emacsdir=#{share}/emacs/site-lisp/gambit-scheme"
   end
 
   test do
```

**Comment by nijikon** [comment]
thanks @dunn & @DomT4

**Comment by DomT4** [comment]
The build timed out here, but passed on Yosemite. I'll bump the timeout and set it running again.

**Comment by nijikon** [comment]
@DomT4 can you bump the timeout so we rerun and can close this?

**Comment by DomT4** [comment]
I bumped the timeout to over two hours the other night & it still failed :/. I'll reschedule it this evening when there's less bot activity.

**Comment by nijikon** [comment]
Awesome!

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Timeout is now at 4 hours. If you hit it again, it probably isn't going to pass :smile_cat:

**Comment by nijikon** [comment]
It did not pass, what are we going to do with this?

**Comment by mikemcquaid** [comment]
Not sure. Does this update often? If not we could build a new version as a one-off.

**Comment by nijikon** [comment]
@mikemcquaid updates are not regular, depends, from the past I can see that there is a new version every 2 months. https://github.com/feeley/gambit/releases

**Comment by nijikon** [comment]
How can I switch the compiler to gcc? So we don't build this with clang but with gcc.

**Comment by mikemcquaid** [comment]
Use `fails_with`

**Comment by nijikon** [comment]
OK, let's try to build this with gcc.

**Comment by nijikon** [comment]
@mikemcquaid LOL, it worked!

**Comment by nijikon** [comment]
I will remove the comment about clang compiler. It is not needed as we mark it as fails with clang. Is that ok?

**Comment by DomT4** [comment]
Go for it. Kind of curious to see if the bot does the build in 20 minutes again, after timing out at 4+ hours previously :smile_cat:.

**Comment by nijikon** [comment]
@DomT4 Took 24 min on master :dancer:

File: Library/Formula/gambit-scheme.rb [hunk 1]
```diff
@@ -32,7 +34,7 @@ def install
     # Don't enable this when using clang, per configure warning.
     args << "--enable-single-host" unless ENV.compiler == :clang
 
-    args << "--enable-shared" if build.include? "enable-shared"
+    args << "--enable-shared" if build.with? "shared"
 
     if ENV.compiler == :clang
```
Comment (dunn) [inline Library/Formula/gambit-scheme.rb]: Can we remove this block now, and the `unless ENV.compiler == :clang` on line 35?

Comment (servercrunch) [inline Library/Formula/gambit-scheme.rb]: Yes, that is exactly what I was asking for, updating now.

Comment (nijikon) [inline Library/Formula/gambit-scheme.rb]: Yes, that is exactly what I was asking for, updating now.

Comment (dunn) [inline Library/Formula/gambit-scheme.rb]: Oh, I missed @nijikon's comment above.

Thanks for sticking with this, and figuring it out!

Comment (DomT4) [inline Library/Formula/gambit-scheme.rb]: You removed slightly too much here :wink:. We need to keep the:

```ruby
args << "--enable-single-host" unless ENV.compiler == :clang
```
line & explanation, as obviously, we'll always be using something that isn't `:clang` now.

If you want to you can compress it into the `args = %W[` block.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/7c8344be9f3249f4c0ea32272f981fbc38639ffe. Thank you for sticking with us on this bumpy ride @nijikon! :smile_cat:

**Pull request closed without merging**